### PR TITLE
Allow duplicate item names

### LINKED ISSUE #5: Allow duplicate item names

### PR DESCRIPTION
**Pull request by alanishungry**
Problem
=======
When adding a item with the same name as an
existing item, nothing happens.

Solution
=======
Array containing all items were indexed by
item name. I have changed them to be indexed
by item id, which theoretically should be a
unique id per item added.

Testing
=======
I made sure the rest of the app works and
made sure item names everywhere were changed
to be retrieved via item id.

Fixes #5

### CODE REVIEW

File: src/Split.jsx [hunk 1]
```diff
@@ -126,18 +126,19 @@ class SplitComponent extends React.Component {
         return this.props.people[index].total
     }
 
-    personItem = (person, itemName) => {
-        if (this.props.items[itemName].people.indexOf(person) >= 0) {
-            var itemPrice = this.props.items[itemName].price
-            var numPeople = this.props.items[itemName].people.length
+    personItem = (person, itemId) => {
+        if (this.props.items[itemId].people.indexOf(person) >= 0) {
+            var itemPrice = this.props.items[itemId].price
```
Comment (ricazhang) [inline src/Split.jsx:131]: const when possible

File: src/ItemList.jsx [hunk 2]
```diff
@@ -9,16 +9,16 @@ class ItemListComponent extends React.Component {
...
+        var itemName = this.props.items[itemId].name;
         var pricePerPerson = parseFloat(itemPrice/itemPeople.length).toFixed(2);
         return (
-            <li id={itemName} key={itemName + itemPrice} content={itemName}>
+            <li id={itemId} key={itemId + itemPrice} content={itemId}>
                 { itemName } for ${ itemPrice } <strong>split by</strong> { this.prettyArray(itemPeople, this.props.numPeople ) } (${ pricePerPerson } per person)
-                <div className="clickable remove-item" onClick={this.removeItem.bind(this, itemName)}>Remove Item</div>
+                <div className="clickable remove-item" onClick={this.removeItem.bind(this, itemId)}>Remove Item</div>
             </li>
         )
```
Comment (ricazhang) [inline src/ItemList.jsx:23]: screenshot?

File: src/ItemList.jsx [hunk 3]
```diff
@@ -9,16 +9,16 @@ class ItemListComponent extends React.Component {
...
-        var itemPeople = this.props.items[itemName].people;
-        var itemId = this.props.items[itemName].id;
+        var itemPrice = this.props.items[itemId].price.toFixed(2);
+        var itemPeople = this.props.items[itemId].people;
+        var itemName = this.props.items[itemId].name;
         var pricePerPerson = parseFloat(itemPrice/itemPeople.length).toFixed(2);
         return (
-            <li id={itemName} key={itemName + itemPrice} content={itemName}>
+            <li id={itemId} key={itemId + itemPrice} content={itemId}>
                 { itemName } for ${ itemPrice } <strong>split by</strong> { this.prettyArray(itemPeople, this.props.numPeople ) } (${ pricePerPerson } per person)
```
Comment (ricazhang) [inline src/ItemList.jsx:20]: just itemId should be fine now right?

**Comment by alanishungry** [comment]
billsplit_duplicates1
billsplit_duplicates2

Will change vars to let / const.

**Pull request closed without merging**